### DISCUSSION TITLE
How to import enum into Frontend

### DISCUSSION THREAD

**Discussion by mankinchi** [prisma/prisma#18699] [Q&A]
Hi everyone, as the title states, I'm unable to import enum into Frontend.

This is my setup:
FE: 
* vite@4.1

BE: 
* prisma@4.12

schema.prisma
```
enum Question_type {
  TEXT
  PARAGRAPH
  ADDRESS
  DATE
  MULTI
  RADIO
  SIGNATURE
}
```
index-browser.js
```
exports.Question_type = makeEnum({
	TEXT: "TEXT",
	PARAGRAPH: "PARAGRAPH",
        ADDRESS: "ADDRESS",
        DATE: "DATE",
        MULTI: "MULTI",
        RADIO: "RADIO",
        SIGNATURE: "SIGNATURE",
});
```
index.d.ts
```
export const Question_type : {
        TEXT: "TEXT",
	PARAGRAPH: "PARAGRAPH",
        ADDRESS: "ADDRESS",
        DATE: "DATE",
        MULTI: "MULTI",
        RADIO: "RADIO",
        SIGNATURE: "SIGNATURE",
};

export type Question_type  = (typeof Question_type )[keyof typeof Question_type ]
```

front-end-script.ts
```
import { Question_type } from "../../server/prisma/generated";

console.log(Question_type );

```

This is the error I got 
`Uncaught SyntaxError: The requested module '/xxxxxxx/server/prisma/generated/index-browser.js?t=1681210470569' does not provide an export named 'Question_type'`

I have tried with both default and custom output for prisma generating. Also, I have no issue with importing types, only problem with enum. How should I proceed with it?

**Comment by mankinchi** [comment]
This is a minimal reproduction if anyone needs https://github.com/mankinchi/minimal-prisma-enum

**Reply by ludralph** [reply]
Hi @mankinchi 

I went through th minimal repository but couldn't find the `enum RenderCondition` you specified and the error message for this minimal repository is quite different from what you shared in the question. This is the current error message in the repository you shared
```
caught SyntaxError: The requested module '/@fs/Users/raphaeletim/Documents/minimal-prisma-enum/server/node_modules/@prisma/client/index-browser.js?v=ddcd53ae' does not provide an export named 'Question_type' (at App.tsx:6:10)
```
Is that correct?

**Reply by nurul3101** [reply]
Hello @mankinchi 👋 

To confirm, in this repository, you are able to import types ( Question, QuestionOption etc) without any issue but the error occurs only in importing these enums (QuestionRenderCondition_condition, Question_type), right?

**Reply by mankinchi** [reply]
@ludralph I would be updating them, but it would be the same use case, with both being the enum I’m trying to import

@nurul3101 the problem happens when I tried to import enum to frontend to use as value.

So this is fine:
```
import {Question_Type} from '@prisma';
Const type: Question_Type = 1
```
Eslint might yell at value 1, but compiling will be fine

but this won’t run and ends with an error
```
import {Question_Type} from '@prisma';
Const type = Question_Type.TEXT
```

**Reply by PhilippAlbrecht-KR** [reply]
I ran into the same issues, was there any solution?

**Comment by mankinchi** [comment]
any update on this?

**Comment by gruckion** [comment]
Please address this issue prisma.

**Comment by electrovir** [comment]
I wrote my own generator to solve this for myself: https://www.npmjs.com/package/prisma-frontend

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
